### PR TITLE
permanents backups for mysql (minor delete refactor)

### PR DESCRIPTION
**Pull request by reshke**

### CODE REVIEW

File: internal/wal_show_handler.go [hunk 1]
```diff
@@ -113,12 +113,12 @@ func (seq *WalSegmentsSequence) FindMissingSegments() ([]WalSegmentDescription,
...
 }
 
-// HandleWalShow gets the list of files inside WAL folder, detects the available WAL segments,
+// HandleWalShow gets the list of files inside WAL Folder, detects the available WAL segments,
 // groups WAL segments by the timeline and shows detailed info about each timeline stored in storage
 func HandleWalShow(rootFolder storage.Folder, showBackups bool, outputWriter WalShowOutputWriter) {
 	walFolder := rootFolder.GetSubFolder(utility.WalPath)
 	filenames, err := getFolderFilenames(walFolder)
-	tracelog.ErrorLogger.FatalfOnError("Failed to get the WAL folder filenames %v\n", err)
+	tracelog.ErrorLogger.FatalfOnError("Failed to get the WAL Folder filenames %v\n", err)
```
Comment (usernamedt) [inline internal/wal_show_handler.go:121]: Looks like an accidental rename

File: cmd/mysql/delete.go [hunk 2]
```diff
@@ -44,22 +44,41 @@ var deleteEverythingCmd = &cobra.Command{
...
+	permanentObjects map[string]bool
+}
+
 func runDeleteEverything(cmd *cobra.Command, args []string) {
-	deleteHandler, err := newMySqlDeleteHandler()
+	deleteHandler, err := NewMySqlDeleteHandler()
 	tracelog.ErrorLogger.FatalOnError(err)
 
+	if p := deleteHandler.permanentObjects; len(p) > 0 {
+		tracelog.InfoLogger.Printf("found permanent objects %s\n", func() string {
```
Comment (usernamedt) [inline cmd/mysql/delete.go:57]: Maybe consider converting to slice and then strings.Join(obj, ",")?

File: cmd/mysql/delete.go [hunk 3]
```diff
@@ -44,22 +44,41 @@ var deleteEverythingCmd = &cobra.Command{
...
+			ret := ""
+
+			for e := range p {
+				ret += e
+				ret += ","
+			}
+
+			return ret
+		}())
+		os.Exit(1)
```
Comment (usernamedt) [inline cmd/mysql/delete.go:67]: tracelog.ErrorLogger.Fatalf?

File: cmd/mysql/delete.go [hunk 4]
```diff
@@ -143,3 +133,66 @@ func tryFetchBinlogName(folder storage.Folder, object storage.Object) (string, b
...
+		}
+		if meta.IsPermanent {
+			permanentBackups[backupTime.BackupName] = true
+		}
+	}
+	return permanentBackups
+}
+
+func IsPermanent(objectName string, permanentBackups map[string]bool) bool {
+	if objectName[:len(utility.BaseBackupPath)] == utility.BaseBackupPath {
```
Comment (usernamedt) [inline cmd/mysql/delete.go:165]: What about permanent binlogs?

File: internal/mark.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type MarkHandler struct {
+	Mark func(uploader *Uploader, folder storage.Folder, backupName string, toPermanent bool)
+}
+
+func (m *MarkHandler) HandleBackupMark(uploader *Uploader, backupName string, toPermanent bool) {
+	folder := uploader.UploadingFolder
```
Comment (usernamedt) [inline internal/mark.go:13]: Most of the MarkHandler logic seems to be done by external Mark function, maybe consider different decomposition?

**Review by usernamedt — CHANGES_REQUESTED** [review]

File: cmd/mysql/delete.go (hunk 2 above)
Comment (reshke) [reply cmd/mysql/delete.go:57]: ok

File: cmd/mysql/delete.go (hunk 4 above)
Comment (reshke) [reply cmd/mysql/delete.go:165]: lets do it in another pr

File: internal/wal_show_handler.go (hunk 1 above)
Comment (reshke) [reply internal/wal_show_handler.go]: fxd

File: internal/databases/mysql/backup_mark_handler.go [hunk 6]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // MarkBackup marks a backup as permanent or impermanent
-func markBackup(uploader *internal.Uploader, folder storage.Folder, backupName string, toPermanent bool) {
+func MarkBackup(uploader *internal.Uploader, folder storage.Folder, backupName string, toPermanent bool) {
 	tracelog.InfoLogger.Printf("Retrieving previous related backups to be marked: toPermanent=%t", toPermanent)
 
 	backup := internal.NewBackup(folder, backupName)
```
Comment (usernamedt) [inline internal/databases/mysql/backup_mark_handler.go:13]: What if the backup with backupName name does not exist in storage?

File: internal/databases/mysql/backup_mark_handler.go [hunk 7]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // MarkBackup marks a backup as permanent or impermanent
-func markBackup(uploader *internal.Uploader, folder storage.Folder, backupName string, toPermanent bool) {
+func MarkBackup(uploader *internal.Uploader, folder storage.Folder, backupName string, toPermanent bool) {
```
Comment (usernamedt) [inline internal/databases/mysql/backup_mark_handler.go:10]: Maybe rename to baseBackupFolder for clarity?

File: cmd/mysql/backup_mark.go [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+	backupMarkShortDescription = "mark permanent/impermanent target backup"
+)
+
+var (
+	// backupMarkCmd represents the backupMark command
+	backupMarkCmd = &cobra.Command{
+		Use:   "backup-mark",
+		Short: BackupMarkShortDescription,
+		Long:  BackupMarkLongDescription,
+		Args:  cobra.ExactArgs(0),
```
Comment (usernamedt) [inline cmd/mysql/backup_mark.go:26]: Arguments count doesn't match the Postgres backup-mark, is it intended?

File: internal/bundle.go [hunk 9]
```diff
@@ -210,7 +210,7 @@ func (bundle *Bundle) HandleWalkedFSObject(path string, info os.FileInfo, err er
 		return nil
 	}
 
-	// Resolve symlinks for tablespaces and save folder structure.
+	// Resolve symlinks for tablespaces and save Folder structure.
```
Comment (usernamedt) [inline internal/bundle.go:213]: still some Folder renamings in comments

**Review by usernamedt — CHANGES_REQUESTED** [review]

File: test/mocks/mock_folder.go [hunk 10]
```diff
@@ -1,5 +1,5 @@
 // Code generated by MockGen. DO NOT EDIT.
-// Source: github.com/wal-g/storages/storage (interfaces: Folder)
+// Source: github.com/wal-g/storages/storage (interfaces: folder)
```
Comment (usernamedt) [inline test/mocks/mock_folder.go:2]: Last one :)

**Review by usernamedt — APPROVED** [review]

**Pull request merged**